src/components/common/MapDateSelect/DateSlider.tsx

```diff
@@ -0,0 +1,44 @@
+import { Slider } from '@terraware/web-components';
+import { useMemo } from 'react';
+
+type DateSliderProps = {
+  dates: Date[];
+  onChange: (newDate: Date) => void;
+  initialSelectionIndex: number;
+};
+
+export default function DateSlider(props: DateSliderProps): JSX.Element {
+  const { dates, onChange, initialSelectionIndex } = props;
+
+  const minTime = useMemo(() => {
+    return Math.min(...dates.map((d) => d.getTime()));
+  }, [dates]);
+
+  const maxTime = useMemo(() => {
+    return Math.max(...dates.map((d) => d.getTime()));
+  }, [dates]);
+
+  const marks = useMemo(() => {
+    return dates.map((date) => ({ value: date.getTime() }));
+  }, [dates]);
+
+  const timestampForDate = useMemo(() => {
+    const tsMap = new Map<number, Date>();
+    for (const date of dates) {
+      tsMap.set(date.getTime(), date);
+    }
+
+    return tsMap;
+  }, [dates]);
+
+  return (
+    <Slider
+      defaultValue={marks[initialSelectionIndex].value}
+      min={minTime}
+      max={maxTime}
```

Comment:
probably don't need this since we allow the entire range of marks as valid values?

Reply:
I started without these, but it wasn't displaying correctly 🤔

Reply:
ok, we can leave it in and figure it out later maybe :)

Reply:
I thought it might be because the marks were sorted backwards, but switching that didn't help either